### PR TITLE
Spring 06

### PR DESCRIPTION
**Pull request by vremere-a**

### CODE REVIEW

File: src/main/java/com/dev/cinema/controllers/CinemaHallController.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Autowired
+    public CinemaHallController(CinemaHallService cinemaHallService,
+                                CinemaHallDtoMapper cinemaHallDtoMapper) {
+        this.cinemaHallService = cinemaHallService;
+        this.cinemaHallDtoMapper = cinemaHallDtoMapper;
+    }
+
+    @GetMapping
+    @ResponseBody
```
Comment (KostiantynVorobiov) [inline src/main/java/com/dev/cinema/controllers/CinemaHallController.java]: эта анотация сдесь лишняя

File: src/main/java/com/dev/cinema/security/implementations/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -24,7 +24,7 @@ public AuthenticationServiceImpl(UserService userService,
 
     @Override
     public User login(String email, String password) throws AuthenticationException {
-        Optional<User> userFromDB = userService.findByEmail(email);
+        Optional<User> userFromDB = Optional.ofNullable(userService.findByEmail(email));
```
Comment (KostiantynVorobiov) [inline src/main/java/com/dev/cinema/security/implementations/AuthenticationServiceImpl.java]: Я думаю не нужно запихивать user в Optional, мы ж его оттуда достали в userService

**Review by KostiantynVorobiov — APPROVED** [review]

File: src/main/java/com/dev/cinema/controllers/CinemaHallController.java (hunk 1 above)
Comment (KseniiaMakarova) [reply src/main/java/com/dev/cinema/controllers/CinemaHallController.java]: I agree, let's remove this annotation

File: src/main/java/com/dev/cinema/controllers/OrderController.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                           UserService userService,
+                           OrderDtoMapper orderDtoMapper) {
+        this.orderService = orderService;
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+        this.orderDtoMapper = orderDtoMapper;
+    }
+
+    @PostMapping("/complete")
+    public void completeOrder(@RequestParam("userId") Long userId) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controllers/OrderController.java]:
```suggestion
    public void completeOrder(@RequestParam Long userId) {
```
it is redundant if you pass the identically named param

File: src/main/java/com/dev/cinema/controllers/OrderController.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        orderService.completeOrder(
+                shoppingCartService.getByUser(
+                        userService.findById(userId)).getTickets(),
+                userService.findById(userId));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controllers/OrderController.java]: let's save User to the variable

File: src/main/java/com/dev/cinema/controllers/OrderController.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @GetMapping
+    public List<OrderResponseDto> getOrderHistory(@RequestParam("userId") Long userId) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controllers/OrderController.java]:
```suggestion
    public List<OrderResponseDto> getOrderHistory(@RequestParam Long userId) {
```

File: src/main/java/com/dev/cinema/controllers/UserController.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    private final UserDtoMapper userDtoMapper;
+
+    @Autowired
+    public UserController(UserService userService, UserDtoMapper userDtoMapper) {
+        this.userService = userService;
+        this.userDtoMapper = userDtoMapper;
+    }
+
+    @GetMapping("/by-email")
+    public UserResponseDto getUserByEmail(@RequestParam("email") String email) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controllers/UserController.java]:
```suggestion
    public UserResponseDto getUserByEmail(@RequestParam String email) {
```

File: src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java [hunk 7]
```diff
@@ -59,4 +59,16 @@ public User add(User user) {
...
+
+    @Override
+    public Optional<User> findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<User> criteriaQuery = criteriaBuilder.createQuery(User.class);
+            Root<User> userEmailRoot = criteriaQuery.from(User.class);
+            Predicate predicate = criteriaBuilder.equal(userEmailRoot.get("id"), id);
+            criteriaQuery.select(userEmailRoot).where(predicate);
+            return session.createQuery(criteriaQuery).uniqueResultOptional();
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java]: let's use `session.get()`

File: src/main/java/com/dev/cinema/model/dto/cart/ShoppingCartResponseDto.java [hunk 8]
```diff
@@ -0,0 +1,10 @@
+package com.dev.cinema.model.dto.cart;
+
+import java.util.List;
+import lombok.Data;
+
+@Data
+public class ShoppingCartResponseDto {
+    private Long cartId;
+    private List<Long> ticketsId;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/cart/ShoppingCartResponseDto.java]:
```suggestion
    private List<Long> ticketIds;
```

File: src/main/java/com/dev/cinema/model/dto/cart/ShoppingCartDtoMapper.java [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+import com.dev.cinema.model.Ticket;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ShoppingCartDtoMapper {
+    public ShoppingCartResponseDto mapToResponseDto(ShoppingCart shoppingCart) {
+        ShoppingCartResponseDto cartResponseDto = new ShoppingCartResponseDto();
+        cartResponseDto.setCartId(shoppingCart.getUser().getId());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/cart/ShoppingCartDtoMapper.java]:
```suggestion
        cartResponseDto.setCartId(shoppingCart.getId());
```
they hold the same values, but anyways

File: src/main/java/com/dev/cinema/model/dto/order/OrderResponseDto.java [hunk 10]
```diff
@@ -0,0 +1,13 @@
+package com.dev.cinema.model.dto.order;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import lombok.Data;
+
+@Data
+public class OrderResponseDto {
+    private Long id;
+    private List<Long> ticketsId;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/order/OrderResponseDto.java]:
```suggestion
    private List<Long> ticketIds;
```

File: src/main/java/com/dev/cinema/model/dto/user/UserRequestDto.java [hunk 11]
```diff
@@ -0,0 +1,9 @@
+package com.dev.cinema.model.dto.user;
+
+import lombok.Data;
+
+@Data
+public class UserRequestDto {
+    private String password;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/user/UserRequestDto.java:7]: do you use this dto?

File: src/main/java/com/dev/cinema/model/dto/user/UserResponseDto.java [hunk 12]
```diff
@@ -0,0 +1,10 @@
+package com.dev.cinema.model.dto.user;
+
+import lombok.Data;
+
+@Data
+public class UserResponseDto {
+    private Long id;
+    private String password;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/user/UserResponseDto.java]: let's not pass secure data

File: src/main/java/com/dev/cinema/security/implementations/AuthenticationServiceImpl.java (hunk 2 above)
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/security/implementations/AuthenticationServiceImpl.java]: inside `findByEmail()` method in service you do the following:

`userDao.findByEmail(email).get();`

what happens if we call optional.get() on empty optional? will we return null or what?

File: src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java [hunk 13]
```diff
@@ -25,7 +24,12 @@ public User add(User user) {
     }
 
     @Override
-    public Optional<User> findByEmail(String email) {
-        return userDao.findByEmail(email);
+    public User findByEmail(String email) {
+        return userDao.findByEmail(email).get();
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java]:
```suggestion
        return userDao.findByEmail(email).orElse(null);
```

I would recommend to do this
and replace `isPresent()` check with null check in AuthenticationServiceImpl

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/controllers/CinemaHallController.java (hunk 1 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/controllers/CinemaHallController.java]: done

File: src/main/java/com/dev/cinema/controllers/OrderController.java (hunk 3 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/controllers/OrderController.java]: done

File: src/main/java/com/dev/cinema/controllers/OrderController.java (hunk 4 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/controllers/OrderController.java]: done

File: src/main/java/com/dev/cinema/controllers/OrderController.java (hunk 5 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/controllers/OrderController.java]: done

File: src/main/java/com/dev/cinema/controllers/UserController.java (hunk 6 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/controllers/UserController.java]: done

File: src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java (hunk 7 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java]: done

File: src/main/java/com/dev/cinema/model/dto/cart/ShoppingCartResponseDto.java (hunk 8 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/model/dto/cart/ShoppingCartResponseDto.java]: done

File: src/main/java/com/dev/cinema/model/dto/cart/ShoppingCartDtoMapper.java (hunk 9 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/model/dto/cart/ShoppingCartDtoMapper.java]: done

File: src/main/java/com/dev/cinema/model/dto/order/OrderResponseDto.java (hunk 10 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/model/dto/order/OrderResponseDto.java]: done

File: src/main/java/com/dev/cinema/model/dto/user/UserRequestDto.java (hunk 11 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/model/dto/user/UserRequestDto.java:7]: yes. in AuthenticationController

File: src/main/java/com/dev/cinema/model/dto/user/UserResponseDto.java (hunk 12 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/model/dto/user/UserResponseDto.java]: done

File: src/main/java/com/dev/cinema/security/implementations/AuthenticationServiceImpl.java (hunk 2 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/security/implementations/AuthenticationServiceImpl.java]: done

File: src/main/java/com/dev/cinema/security/implementations/AuthenticationServiceImpl.java [hunk 14]
```diff
@@ -24,9 +24,8 @@ public AuthenticationServiceImpl(UserService userService,
 
     @Override
     public User login(String email, String password) throws AuthenticationException {
-        Optional<User> userFromDB = Optional.ofNullable(userService.findByEmail(email));
-        if (userFromDB.isPresent()
-                && userFromDB.get().getPassword().equals(HashUtil.hashPassword(password,
+        Optional<User> userFromDB = userService.findByEmail(email);
+        if (userFromDB.get().getPassword().equals(HashUtil.hashPassword(password,
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/security/implementations/AuthenticationServiceImpl.java]: let's return `isPresent()` check

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/security/implementations/AuthenticationServiceImpl.java (hunk 14 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/security/implementations/AuthenticationServiceImpl.java]: ok. done

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**